### PR TITLE
Make bucket Integration Test

### PR DESCRIPTION
**Pull request by cniackz**
To add a new test where we create or add a bucket testing with `locking` as `true` since that hasn't been tested yet. We need to make sure we can create the bucket even with this field as `true` 

Fixes miniohq/engineering#374

### CODE REVIEW

**Review by bexsoft — DISMISSED** [review]
LGTM

File: integration/buckets_test.go [hunk 1]
```diff
@@ -208,6 +209,108 @@ func TestAddBucket(t *testing.T) {
...
+		"name":       "test1",
+		"versioning": false,
+		"locking":    true,
+	}
+
+	requestDataJSON, _ := json.Marshal(requestDataAdd)
+
+	requestDataBody := bytes.NewReader(requestDataJSON)
+
+	// get list of buckets
```
Comment (prakashsvmx) [inline integration/buckets_test.go:239]: this code block creates a bucket so the comment can be updated accordingly.

```suggestion
	// create a object locking bucket without versioning flag
```

File: integration/buckets_test.go [hunk 2]
```diff
@@ -208,6 +209,108 @@ func TestAddBucket(t *testing.T) {
...
+		"name":       "thujan",
+		"versioning": true,
+		"locking":    true,
+	}
+
+	requestDataJSON, _ = json.Marshal(requestDataAdd)
+
+	requestDataBody = bytes.NewReader(requestDataJSON)
+
+	// get list of buckets
```
Comment (prakashsvmx) [inline integration/buckets_test.go:277]: Please update the comment here

File: integration/buckets_test.go [hunk 3]
```diff
@@ -208,6 +209,108 @@ func TestAddBucket(t *testing.T) {
...
+	request.Header.Add("Content-Type", "application/json")
+
+	response, err = client.Do(request)
+	assert.Nil(err)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	// Verification part, bucket name is valid, we expect 201 when created.
```
Comment (prakashsvmx) [inline integration/buckets_test.go:295]:
```suggestion
	// Verification part, bucket should be created with versioning enabled and locking enabled, we expect 201 when created.
```

**Review by prakashsvmx — APPROVED** [review]
minor code comment adjustments
otherwise
Changes look good to me :+1:

File: integration/buckets_test.go (hunk 1 above)
Comment (cniackz) [reply integration/buckets_test.go]: The change has been performed.
Thank you for the suggestion @prakashsvmx

File: integration/buckets_test.go (hunk 2 above)
Comment (cniackz) [reply integration/buckets_test.go]: Also this comment was updated accordingly

File: integration/buckets_test.go (hunk 3 above)
Comment (cniackz) [reply integration/buckets_test.go]: That is right, this is not about the name but the combination of versioning and locking when creating the bucket, thanks!

**Review by prakashsvmx — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**